pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +999,38 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// check whether zone endpoint list need to be updated or not
+func UpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) bool {
+	updateZoneEndpointList := false
+	isZonePresent := false
+	zoneEndpointsList := []string{}
+	checkZoneEndpoints := make(map[string]bool)
+	if zoneName != "" {
+		if len(zones) > 0 {
+			for _, z := range zones {
+				if z.Name == zoneName {
+					for _, endpoint := range z.Endpoints {
+						zoneEndpointsList = append(zoneEndpointsList, endpoint)
+						checkZoneEndpoints[endpoint] = true
+					}
+					isZonePresent = true
+					break
+				}
+			}
+		}
+		if isZonePresent {
+			if len(desiredEndpointList) != len(zoneEndpointsList) {
+				updateZoneEndpointList = true
+			} else {
+				for _, endpoint := range desiredEndpointList {
+					if !checkZoneEndpoints[endpoint] {
+						updateZoneEndpointList = true
+						break
+					}
+				}
+			}
```

Comment:
I can see what is happening here with the list and map, and it is unnecessarily complex. It is trivial to determine if two lists are the same by comparing length and using a sorting function. See here: https://go.dev/play/p/5GJH6JmaOoY

With this, you can write a function that will tell you if two endpoint lists are the same. Said function can be unit tested to verify its correctness.